pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -50,16 +144,31 @@ func (r *Reconciler) create() error {
 		CanIpForward:       r.providerSpec.CanIPForward,
 		DeletionProtection: r.providerSpec.DeletionProtection,
 		Labels:             r.providerSpec.Labels,
-		MachineType:        fmt.Sprintf("zones/%s/machineTypes/%s", zone, r.providerSpec.MachineType),
+		MachineType:        fmt.Sprintf(machineTypeFmt, zone, r.providerSpec.MachineType),
 		Name:               r.machine.Name,
 		Tags: &compute.Tags{
 			Items: r.providerSpec.Tags,
 		},
 		Scheduling: &compute.Scheduling{
-			Preemptible: r.providerSpec.Preemptible,
+			Preemptible:       r.providerSpec.Preemptible,
+			AutomaticRestart:  r.providerSpec.AutomaticRestart,
+			OnHostMaintenance: r.providerSpec.OnHostMaintenance,
 		},
 	}
 
+	var guestAccelerators = []*compute.AcceleratorConfig{}
+	for index, ga := range r.providerSpec.GuestAccelerators {
+		guestAccelerators = append(guestAccelerators, &compute.AcceleratorConfig{
+			AcceleratorType:  fmt.Sprintf(acceleratorTypeFmt, zone, r.providerSpec.GuestAccelerators[index].AcceleratorType),
```

Comment:
My mistake. I assumed since `r.providerSpec.GuestAccelerators` is slice, it might have more then one `AcceleratorConfig`s in it. However it cannot have more than one. 

Same logic applied [here](https://github.com/openshift/cluster-api-provider-gcp/pull/172/files#diff-359c93fbc08f569c3abf79881f0b426e681c46a5e76765d05e921f16fdb3bf99R79).